### PR TITLE
OOP school library: set up associations

### PR DESCRIPTION
**Pull request by shubhambhoyar077**
## In this PR:-

Created a class called Classroom.

Created the has-many/belongs-to relationship between Classroom and Student. 

Created a class called Book.

Created a class called Rental.

Created the many-to-many (also has-many-through) relationship between Person and Book using the intermediate class Rental.

### CODE REVIEW

File: book.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+class Book
+  attr_accessor :title, :author
+  attr_reader :rentals
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
+end
```
Comment (GzimAsani) [inline book.rb:10]: - [ ] In order to create many-to-many (also has-many-through) relationship between Person and Book using the intermediate class Rental you need to create a method `add_rental`, something similar to 

```
  def add_rental(person, date)
    *the book itself should be sent as a parameter to create the new rental achieving the 'has-many' association. 
    Rental.new(date, self, person)
  end
```

Btw the rental class is looking fine

File: book.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+class Book
+  attr_accessor :title, :author
+  attr_reader :rentals
```
Comment (GzimAsani) [inline book.rb:3]: - [ ] Add :rentals as accessors since you need to get and set it

File: person.rb [hunk 3]
```diff
@@ -4,14 +4,15 @@
 
 class Person < Nameable
   attr_accessor :name, :age
-  attr_reader :id
+  attr_reader :id, :rentals
```
Comment (GzimAsani) [inline person.rb:7]: - [ ] Add :rentals as attr_accessor here, since you need to use it as a setter as well

File: person.rb [hunk 4]
```diff
@@ -4,14 +4,15 @@
...
   def initialize(age, name: 'Unknown', parent_permission: true)
     super()
     @id = 0
     @name = name
     @age = age
     @parent_permission = parent_permission
+    @rentals = []
   end
 
   def correct_name
```
Comment (GzimAsani) [inline person.rb:18]: - [ ] Add a method here to for adding rentals, example:

```ruby
  def add_rental(book, date)
    # *The person itself should be sent as a parameter to create the new rental achieving the 'has-many' association
    Rental.new(date, book, self)
  end
```

**Review by GzimAsani — CHANGES_REQUESTED** [review]
### Hello, @shubhambhoyar077              :wave:   
 Good job so far,  done There are some issues that you still need to work on to go to the next project but you are almost there!
 
 ## Highlights
 * Good job with creating all the necessary classes :+1: 

 ## Required Changes 

- [ ] Check the comments below :arrow_double_down: 

 ## Optional suggestions
 _Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding! clapclapclap
 
 Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.
 
 **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
 
 _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: book.rb (hunk 1 above)
Comment (shubhambhoyar077) [reply book.rb]: Hi @GzimAsani,
Thanks for review. :smile: 
Added add_rental method in [book class](https://github.com/shubhambhoyar077/oop-school-library/pull/3/commits/66a3b375518700c140a39c8949fdfec36d674f06)

File: book.rb (hunk 2 above)
Comment (shubhambhoyar077) [reply book.rb]: Updated rentals to [accessors](https://github.com/shubhambhoyar077/oop-school-library/pull/3/commits/77e84872977ac5c6d0184f5b0ea88459a6ae9c82)

File: person.rb (hunk 3 above)
Comment (shubhambhoyar077) [reply person.rb]: Updated rentals to [accessors](https://github.com/shubhambhoyar077/oop-school-library/pull/3/commits/6f1b6c187e06779a795a86ec7a02364968df351d)

File: person.rb (hunk 4 above)
Comment (shubhambhoyar077) [reply person.rb]: Added add_rentals method in [person class]([4bd0ac6](https://github.com/shubhambhoyar077/oop-school-library/pull/3/commits/4bd0ac68cb887694ecf02f70e7c738c9ad94e21d))

**Review by julie-ify — APPROVED** [review]
Hi @shubhambhoyar077,

## Highlights!
- Good code organization and working app ✔️
- Descriptive README ✔️ 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# Status: Approved ✔️ 

Cheers and Happy coding!👏👏👏

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: rental.rb [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+  attr_accessor :date, :book, :person
+
+  def initialize(book, person, date)
+    @date = date
+
+    @book = book
+    @book.rentals << self
+
+    @person = person
+    @person.rentals << self
```
[the remark this answers is not in the document]
Comment (shubhambhoyar077) [reply rental.rb]: Hi @julie-ify ,
Thanks for approval. :smile: 
Above issue is fixed :+1:

**Pull request merged**